packages/zoe/src/typeGuards.js

```diff
@@ -79,3 +121,243 @@ export const InvitationElementShape = M.splitRecord({
 export const OfferHandlerI = M.interface('OfferHandler', {
   handle: M.call(SeatShape).optional(M.any()).returns(M.string()),
 });
+
+export const SeatHandleAllocationsShape = M.arrayOf(
+  harden({
+    seatHandle: SeatShape,
+    allocation: AmountKeywordRecordShape,
+  }),
+);
+
+export const ZoeMintI = M.interface('ZoeMint', {
+  getIssuerRecord: M.call().returns(IssuerRecordShape),
+  mintAndEscrow: M.call(AmountShape).returns(),
+  withdrawAndBurn: M.call(AmountShape).returns(),
+});
+
+export const ZcfMintI = M.interface('ZcfMint', {
+  getIssuerRecord: M.call().returns(IssuerRecordShape),
+  mintGains: M.call(AmountKeywordRecordShape, M.remotable('zcfSeat')).returns(),
+  burnLosses: M.call(
+    AmountKeywordRecordShape,
+    M.remotable('zcfSeat'),
+  ).returns(),
+});
+
+export const ExitObjectI = M.interface('Exit Object', {
+  exit: M.call().returns(),
+});
+
+const ExitObjectShape = M.remotable('ExitObj');
+export const InstanceAdminShape = M.remotable('InstanceAdmin');
+export const InstanceAdminI = M.interface('InstanceAdmin', {
+  makeInvitation: M.call(InvitationHandleShape, M.string())
+    .optional(M.record(), M.pattern())
+    .returns(M.promise()),
+  saveIssuer: M.callWhen(M.await(IssuerShape), M.string()).returns(
+    IssuerRecordShape,
+  ),
+  makeNoEscrowSeat: M.call(
+    AmountKeywordRecordShape,
+    ProposalShape,
+    ExitObjectShape,
+    SeatShape,
+  ).returns(SeatShape),
+  exitAllSeats: M.call(M.any()).returns(),
+  failAllSeats: M.call(M.any()).returns(),
+  exitSeat: M.call(SeatShape, M.any()).returns(),
+  failSeat: M.call(SeatShape, M.any()).returns(),
+  makeZoeMint: M.call(KeywordShape)
+    .optional(AssetKindShape, DisplayInfoShape, M.pattern())
+    .returns(M.remotable('zoeMint')),
+  registerFeeMint: M.call(KeywordShape, M.remotable('feeMintAccess')).returns(
+    M.remotable('feeMint'),
+  ),
+  replaceAllocations: M.call(SeatHandleAllocationsShape).returns(),
+  stopAcceptingOffers: M.call().returns(),
+  setOfferFilter: M.call(M.arrayOf(M.string())).returns(),
+  getOfferFilter: M.call().returns(M.arrayOf(M.string())),
+  getExitSubscriber: M.call(SeatShape).returns(SubscriberShape),
+  isBlocked: M.call(M.string()).returns(M.boolean()),
+});
+
+export const InstanceStorageManagerIKit = harden({
+  instanceStorageManager: M.interface('InstanceStorageManager', {
+    getTerms: M.call().returns(M.split(TermsShape, M.record())),
+    getIssuers: M.call().returns(IssuerKeywordRecordShape),
+    getBrands: M.call().returns(BrandKeywordRecordShape),
+    getInstallationForInstance: M.call().returns(InstallationShape),
+    getInvitationIssuer: M.call().returns(IssuerShape),
+
+    saveIssuer: M.call(IssuerShape, M.string()).returns(M.promise()),
+    makeZoeMint: M.call(KeywordShape)
+      .optional(AssetKindShape, DisplayInfoShape, M.pattern())
+      .returns(M.or(ZoeMintI, M.remotable('zoeMint'), M.promise())),
```

Comment:
`ZoeMintI` is presumably an interface guard, which is therefore not a pattern, and so should not appear here.

Do you already have a `ZoeMintShape` that's identical to `M.remotable('ZoeMint')`? If so, should this all be simplified to
```suggestion
      .returns(M.eref(ZoeMintShape)),
```

Reply:
done